clusterloader2/pkg/measurement/util/wait_for_conditions.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+)
+
+// WaitForGenericK8sObjectsOptions is an options object used by WaitForGenericK8sObjectsNodes methods.
+type WaitForGenericK8sObjectsOptions struct {
+	// GroupVersionResource identifies the resource to fetch.
+	GroupVersionResource schema.GroupVersionResource
+	// Namespaces identifies namespaces which should be observed.
+	Namespaces NamespacesRange
+	// SuccessfulConditions lists conditions to look for in the objects denoting good objects.
+	// Formatted as `ConditionType=ConditionStatus`, e.g. `Scheduled=true`.
+	SuccessfulConditions []string
+	// SuccessfulConditions lists conditions to look for in the objects denoting good objects.
+	// Formatted as `ConditionType=ConditionStatus`, e.g. `Scheduled=true`.
+	FailedConditions []string
+	// MinDesiredObjectCount describes minimum number of objects that should contain
+	// successful or failed condition.
+	MinDesiredObjectCount int
+	// MaxFailedObjectCount describes maximum number of objects that could contain failed condition.
+	MaxFailedObjectCount int
+	// CallerName identified ther caller.
```

Comment:
Who is the caller in this context?

Reply:
This value contains the name of the measurement that is using this function.
Updated the comment to reflect this.